### PR TITLE
Fix default relationship for existing contact element

### PR DESCRIPTION
**Pull request by jitendrapurohit**
Overview
----------------------------------------
Fix contact default relationship options. 

Drupal Issue - https://www.drupal.org/project/webform_civicrm/issues/3195941

Before
----------------------------------------
Default section on the existing contact element is broken. 

image

After
----------------------------------------
Fixed.

image

Comments
----------------------------------------
@KarinG

### CODE REVIEW

File: src/ContactComponent.php [hunk 1]
```diff
@@ -177,6 +177,7 @@ function wf_crm_contact_access($component, $filters, $cid) {
     $utils = \Drupal::service('webform_civicrm.utils');
     // Create new contact doesn't require lookup
     $cid = (string) $cid;
+    $component['#form_key'] = $component['#form_key'] ?? $component['#webform_key'];
     list(, $c, ) = explode('_', $component['#form_key'], 3);
```
Comment (KarinG) [inline src/ContactComponent.php:181]: Our #form_key-s are all over the place. Definitively a ToDo item to really get to the bottom of #form_key-s between a final beta5 release and a full release.

**Comment by KarinG** [comment]
Tests are ✅ 🎉

**Comment by ghost** [comment]
I am still getting the same fault, having inserted the patch, cleared caches and created a new test form.  Configuring Existing Contact on Contact 2 still gives:

image

This is the export of the webform.
[webform.webform.testwf4.yml.zip](https://github.com/colemanw/webform_civicrm/files/5939278/webform.webform.testwf4.yml.zip)

**Comment by KarinG** [comment]
@maynardsmith - what are your Drupal and webform (itself) versions?

**Comment by ghost** [comment]
Drupal 8.9.13,  I was previously working with Webform 6.0.0., but have just upgraded to 6.0.1 with the same result.

**Comment by KarinG** [comment]
Testing this myself as well:

Before: 
image

curl https://patch-diff.githubusercontent.com/raw/colemanw/webform_civicrm/pull/466.diff --output 466.diff
git apply 466.diff
drush cr

After:
image

This PR also includes a test addition and it's passing. I'm going to merge it.

@maynardsmith - happy to help figure out what may be happening on your end - I'll post a yaml comparison.

**Comment by KarinG** [comment]
image

**Comment by KarinG** [comment]
I've even tested this with multiple existing contact elements (and different contact_types and contact sub_types) - and it's all working. Sorry - I can't spot anything in your YAML that looks off. Are you positive all caches are cleared? 

image

**Comment by ghost** [comment]
I would have sworn that I applied that same patch correctly, though I did it by hand.  I have just repeated using the commands you listed above, and it does indeed work.  Thanks for your efforts.

**Comment by KarinG** [comment]
So happy to hear this! And now you have a faster way to apply patches 👍

**Pull request merged**